### ISSUE TITLE
Non-standard error message when registering with a mobile number that's already taken

### ISSUE THREAD

**Issue by jonuy** [DoSomething/northstar#183]
Current error response is something like...
```
mobile =     (
       "The mobile has already been taken."
)
```

The expectation is the error would be something like...
```
{
  error: {
    code: 412,
    message: "Incorrect password."
  }
}
```

Maybe we throw a try/catch block around the validate code in `UserController::store`? I'm assuming the error response is getting triggered by an exception coming out of there

**Comment by DFurnes** [comment]
We should be able to just override the error handler [`buildFailedValidationResponse`](https://github.com/laravel/framework/blob/16f59c2d86eee2e67c7947fd5c7a1a8fafc95ad4/src/Illuminate/Foundation/Validation/ValidatesRequests.php#L74-L90) that's added from the `ValidatesRequests` trait in our `Controller` base class.

**Comment by aaronschachter** [comment]
Any chance who ever takes this on can close out  #172 also?

**Comment by chloealee** [comment]
@jonuy @aaronschachter looks like there is no error message popping up if the mobile number exists already and instead just let's the user overwrite the password like #172 says. I can hop on #172 as well and fix that before this issue/wrap this issue into the fix for #172.
